docs/data/data-grid/server-side-data/tree-data.md

```diff
@@ -2,14 +2,36 @@
 title: React Server-side tree data
 ---
 
-# Data Grid - Server-side tree data [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')🚧
+# Data Grid - Server-side tree data [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
 
 <p class="description">Tree data lazy-loading with server-side data source.</p>
 
-:::warning
-This feature isn't implemented yet. It's coming.
+To use the server-side tree data, pass the `unstable_dataSource` prop as explained in the [overview section](/x/react-data-grid/server-side-data/), in addition to that passing of some additional props is required for the server-side tree data to work properly.
 
-👍 Upvote [issue #3377](https://github.com/mui/mui-x/issues/3377) if you want to see it land faster.
+- `getGroupKey(row: GridRowModel): string`
 
-Don't hesitate to leave a comment on the same issue to influence what gets built. Especially if you already have a use case for this component, or if you are facing a pain point with the [currently proposed workaround](https://mui.com/x/react-data-grid/tree-data/#children-lazy-loading).
+  Used by the grid to group rows by their parent group. Replaces `getTreeDataPath` used in client-side tree-data.
```

Comment:
> will we deprecate getTreeDataPath and use getGroupKey on client side as well?

Sure we can do it, but I am not sure if it will help much with the client data. The motivation for having `getGroupKey` instead of the path on the server data is to avoid computing `path` value for each row with every fetch (it might be an expensive call on the BE side), while with the client data, all the data is already available and computing the path shouldn't be much of an issue.
Or is your concern more around API consistency?